odoo/addons/base/models/ir_ui_view.py

```diff
@@ -370,6 +372,19 @@ def _search_model_data_id(self, operator, value):
         data = self.env['ir.model.data'].sudo().search(domain)
         return [('id', 'in', data.mapped('res_id'))]
 
+    @api.depends('model')
+    def _compute_model_id(self):
+        model_name_id_map = {
+            m.model: m.id
+            for m in self.env['ir.model'].search([('model', 'in', self.mapped('model'))])
+        }
+        for record in self:
+            record.model_id = model_name_id_map.get(record.model)
```

Comment:
There's the method `_get()` on model `ir.model` that does the "name resolution" on models, and even uses a cache for performance:
```suggestion
        for record in self:
            record.model_id = self.env['ir.model']._get(record.model)
```

Reply:
`_get` on `ir.model` searches models with a specific `name`, here I am searching for `ir.model : model` which is also a string, but not the name :)

Reply:
ho no, there is the bad naming and bad comments. It actually searches on model, sorry.

Reply:
Look at the details of the method `_get()`: it actually resolves the model's `_name`, which is exactly what you need here.